File: baselibrary/src/main/java/com/saltwater2233/baselibrary/utils/ToastUtil.java
package com.saltwater2233.baselibrary.utils;

import android.content.Context;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.Toast;

/**
 * <pre>
 *     author : wenxin
 *     e-mail : wenxin2233@outlook.com
 *     time   : 2018/02/26
 *     desc   : Toast统一管理类
 *     version: 1.0
 * </pre>
 */

public class ToastUtil {
    private static Toast toast;
    private static Snackbar snackbar;

    /**
     * 短时间显示Toast
     *
     * @param context
     * @param message
     */
    public static void showShort(Context context, CharSequence message) {
        if (toast == null) {
            toast = Toast.makeText(context, message, Toast.LENGTH_SHORT);
        } else {
            toast.setText(message);
        }
        toast.show();//设置新的消息提示
    }

    /**
     * 短时间显示Toast
     *
     * @param context
     * @param message
     */
    public static void showShort(Context context, int message) {
        if (toast == null) {
            toast = Toast.makeText(context, message, Toast.LENGTH_SHORT);
        } else {
            toast.setText(message);
        }
        toast.show();

    }

    /**
     * 长时间显示Toast
     *
     * @param context
     * @param message
     */
    public static void showLong(Context context, CharSequence message) {
        if (toast == null) {
            toast = Toast.makeText(context, message, Toast.LENGTH_LONG);
        } else {
            toast.setText(message);
        }
        toast.show();

    }

    /**
     * 长时间显示Toast
     *
     * @param context
     * @param message
     */
    public static void showLong(Context context, int message) {
        if (toast == null) {
            toast = Toast.makeText(context, message, Toast.LENGTH_LONG);
        } else {
            toast.setText(message);
        }
        toast.show();
    }

    /**
     * 自定义显示Toast时间
     *
     * @param context
     * @param message
     * @param duration
     */
    public static void show(Context context, CharSequence message, int duration) {
        if (toast == null) {
            toast = Toast.makeText(context, message, duration);
        } else {
            toast.setText(message);
        }
        toast.show();

    }

    /**
     * 自定义显示Toast时间
     *
     * @param context
     * @param message
     * @param duration
     */
    public static void show(Context context, int message, int duration) {
        if (toast == null) {
            toast = Toast.makeText(context, message, duration);
        } else {
            toast.setText(message);
        }
        toast.show();

    }

    /**
     * 显示SnackBar
     *
     * @param view
     * @param listener
     */
    public static void showSnackBarShort(View view, CharSequence text, View.OnClickListener listener) {
        if (snackbar == null) {
            snackbar = Snackbar.make(view, text, Snackbar.LENGTH_SHORT).setAction("撤销", listener);
        } else {
            snackbar.setAction(text, listener);
        }
        snackbar.show();
    }

    public static void showSnackBarLong(View view, CharSequence text, View.OnClickListener listener) {
        if (snackbar == null) {
            snackbar = Snackbar.make(view, text, Snackbar.LENGTH_LONG).setAction("撤销", listener);
        } else {
            snackbar.setAction(text, listener);
        }
        snackbar.show();
    }
}

File: common/src/main/java/com/saltwater/common/utils/ToastUtil.kt
package com.saltwater.common.utils

import android.graphics.drawable.Drawable
import android.widget.Toast
import com.saltwater.common.BaseApplication
import es.dmoral.toasty.Toasty

/**
 * <pre>
 * author : wenxin
 * e-mail : wenxin2233@outlook.com
 * time   : 2018/02/26
 * desc   : Toast统一管理类
 * version: 1.0
</pre> *
 */

object ToastUtil {

    /***
     * 通常的Toast
     */
    fun show(message: Int) {
        Toasty.normal(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    fun show(message: CharSequence) {
        Toasty.normal(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 错误Toast：
     */
    fun showError(message: Int) {
        Toasty.error(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    fun showError(message: CharSequence) {
        Toasty.error(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 成功Toast：
     */
    fun showSuccess(message: Int) {
        Toasty.success(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    fun showSuccess(message: CharSequence) {
        Toasty.success(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 信息Toast：
     */
    fun showInfo(message: Int) {
        Toasty.info(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    fun showInfo(message: CharSequence) {
        Toasty.info(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 警告Toast：
     */
    fun showWarning(message: Int) {
        Toasty.warning(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    fun showWarning(message: CharSequence) {
        Toasty.warning(BaseApplication.context, message, Toast.LENGTH_SHORT).show()
    }

    /**
     * 带有图标的常用Toast：
     */
    fun showIcon(message: Int, icon: Drawable) {
        Toasty.normal(BaseApplication.context, message, icon).show()
    }

}
